addons/udes_stock/models/stock_move_line.py

```diff
@@ -0,0 +1,195 @@
+        """This function computes the different product quantities for the given move_lines
+           :kwargs:
+               - get_key: a callable which takes a quant and returns the key
+           TODO: Change function name - to what?
+           Seems sensible unless we add a get_value option also
+        """
+        res = defaultdict(int)
+        for move_line in self:
+            res[get_key(move_line)] += move_line.product_uom_qty
+        return res
+
+    def sort_by_key(self, sort_key=lambda ml: (ml.location_id.name, ml.product_id.id)):
+        """ Return the move lines sorted by location and product
+        """
+        return self.sorted(key=sort_key)
+
+    def _split(self, qty=0):
+        """ Splits the move line by
+            - qty_done if qty_done > 0
+            - by qty if qty > 0 and qty_done == 0
+            As cannot split by qty if some already done!
+
+            :kwargs:
+                - qty: int
+                    Split current move line in self in two move lines
+                    by quantity if qty > 0, return the move line to left of qty
+            :returns:
+                either self or the new move line
+        TODO: Tidy up function - currently just merged the two split funcitons in udes_stock
+        TODO: Check why the ordered qty get split the opposite way around?
+        """
+        self.ensure_one()
+        res = self
+        qty_done = self.qty_done
+        split_qty = qty_done if qty_done > 0 else qty
+        if (
+            split_qty > 0
+            and float_compare(
+                split_qty, self.product_uom_qty, precision_rounding=self.product_uom_id.rounding
+            )
+            < 0
+        ):
+            # Now we need to map the split properly as the split occurs in the mirror of each other
+            # So we have a left_ml and right ml, where for qty_done the new ml is the right,
+            # else the new ml is the left one
+            # <--------------------x-------------> x = split point
+            # <----ml_left---------x--ml_right--->
+            # <----qty-------------x-------------> If qty_done == 0
+            # <----qty-done---p----x---new_ml----> If p=qty_done > 0, and x = qty > 0, split at p ignore x
+
+            left_ml_qty = split_qty
+            right_ml_qty = float_round(
+                self.product_uom_qty - split_qty,
+                precision_rounding=self.product_uom_id.rounding,
+                rounding_method="UP",
+            )
+
+            # Now compute the left/right ml_ordered_qty
+            # Note in qty_done split they just set right_ml_ordered_qty = right_ml_qty
+            left_ml_ordered_qty = split_qty
+            right_ml_ordered_qty = float_round(
+                self.ordered_qty - split_qty,
+                precision_rounding=self.product_uom_id.rounding,
+                rounding_method="UP",
+            )
+
+            if qty_split > self.ordered_qty:
```

Comment:
ordered_qty no longer in Odoo13
Do we want to use that still so need to re-add it in, or not

Reply:
Removed